microsoft/PowerToys
# [FancyZones] Child windows support

## Summary of the Pull Request

**What is this about:**

Fix child and popup windows support.
Settings and utils refactoring.
Almost everything from utils to window utils was moved as it is, except `IsCandidateForZoning`, `IsStandardWindow`, and new `IsPopupWindow`, `HasThickFrameAndMinimizeMaximizeButtons`.

**What is included in the PR:** 

**How does someone test / validate:** 

* Verify regular window snapping is not affected.
* Verify popup window snapping with setting on/off
* Verify child window snapping with setting on/off.
* Verify start menu can't be snapped with popup setting on/off.
* Verify PT windows such as ColorPicker or PT Run can't be snapped.

While testing if the window can be snapped, try both dragging with `Shift` pressed and keyboard shortcut `Win + Arrow`.

If it's possible, try to snap child windows from the list below.
List of apps where child windows should be snapped:
* Visual Studio
* Notepad++ (Find)
* CAD
* Eclipse
* Autodesk
* Unreal Engine
* 3DS Max
* Meta Trader
* Ninja Trader
* Inkscape

## Quality Checklist

- [x] **Linked issue:** #15426

## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.
